core/model/modx/modx.class.php

```diff
@@ -1267,7 +1267,7 @@ public function getUser($contextKey= '',$forceLoadSettings = false) {
             $this->user = $this->newObject('modUser');
             $this->user->fromArray(array(
                 'id' => 0,
-                'username' => '(anonymous)'
+                'username' => $this->config['anonymous_username']
```

Comment:
Wouldn't it be "safer" to use `$this->getOption('anonymous_username', null, '(anonymous)')` in case the system setting is deleted ? (which would provide a "default" value)